src/ui.rs

```diff
@@ -240,15 +335,49 @@ impl Ui {
     pub fn output_guard(&self, text: String) -> OutputGuard {
         OutputGuard {
             text,
-            output: match self.output_pair {
-                UiOutputPair::Terminal { .. } => io::stdout(),
+            output: match self.output {
+                UiOutput::Terminal { .. } => io::stdout(),
+                UiOutput::Paged { .. } => io::stdout(),
```

Comment:
@Ralith I wasn't absolutely sure what to do here, but since the purpose is to write when it is dropped by the signal handler, it made sense to me to write to `io::stdout()` if we should write at all.

Reply:
I think probably we don't need to do anything at all if the pager currently owns the terminal, since it'll have its own state handling, so this is fine, though we could also come up with a null output by tweaking the `OutputGuard` definition a little.

Reply:
Okay, I'll leave a little TODO comment then